### ISSUE TITLE
DAG `TI_taskflow` fails at local shell because os.path.exist() does not support wildcards

### ISSUE THREAD

**Issue by Feiyang472** [deepmodeling/dpdispatcher#93]
This causes a bug when using airflow to run DAG `TI_taskflow` with local shell.

I checked that python3 does not understand wild cards. 
```python
>>> not os.path.exists("$workdir/200K-50000bar-beta.lmp/NPT_sim/new_job/./*lmp")
True
>>> not os.path.exists("$workdir/200K-50000bar-beta.lmp/NPT_sim/new_job/./beta.lmp")
False
```

Relevant log
```
*** Reading local file: /home/feiyang/airflow/logs/TI_taskflow/NPT_sim/2021-07-09T06:11:43+00:00/1.log
[2021-07-09 14:11:48,439] {taskinstance.py:896} INFO - Dependencies all met for <TaskInstance: TI_taskflow.NPT_sim 2021-07-09T06:11:43+00:00 [queued]>
[2021-07-09 14:11:48,443] {taskinstance.py:896} INFO - Dependencies all met for <TaskInstance: TI_taskflow.NPT_sim 2021-07-09T06:11:43+00:00 [queued]>
[2021-07-09 14:11:48,443] {taskinstance.py:1087} INFO - 
--------------------------------------------------------------------------------
[2021-07-09 14:11:48,443] {taskinstance.py:1088} INFO - Starting attempt 1 of 1
[2021-07-09 14:11:48,443] {taskinstance.py:1089} INFO - 
--------------------------------------------------------------------------------
[2021-07-09 14:11:48,455] {taskinstance.py:1107} INFO - Executing <Task(_PythonDecoratedOperator): NPT_sim> on 2021-07-09T06:11:43+00:00
[2021-07-09 14:11:48,458] {standard_task_runner.py:52} INFO - Started process 17692 to run task
[2021-07-09 14:11:48,461] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'TI_taskflow', 'NPT_sim', '2021-07-09T06:11:43+00:00', '--job-id', '51', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/DpFreeEnergy.py', '--cfg-path', '/tmp/tmpyo02_6_v', '--error-file', '/tmp/tmp0r3tqwl1']
[2021-07-09 14:11:48,461] {standard_task_runner.py:77} INFO - Job 51: Subtask NPT_sim
[2021-07-09 14:11:48,483] {logging_mixin.py:104} INFO - Running <TaskInstance: TI_taskflow.NPT_sim 2021-07-09T06:11:43+00:00 [running]> on host FeiyangChen-Inspiron.localdomain
[2021-07-09 14:11:48,515] {taskinstance.py:1300} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=fengbo
AIRFLOW_CTX_DAG_ID=TI_taskflow
AIRFLOW_CTX_TASK_ID=NPT_sim
AIRFLOW_CTX_EXECUTION_DATE=2021-07-09T06:11:43+00:00
AIRFLOW_CTX_DAG_RUN_ID=manual__2021-07-09T06:11:43+00:00
[2021-07-09 14:11:48,516] {taskinstance.py:1501} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/feiyang/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1157, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/feiyang/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1331, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/feiyang/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1361, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/feiyang/.local/lib/python3.8/site-packages/airflow/decorators/base.py", line 134, in execute
    return_value = super().execute(context)
  File "/home/feiyang/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 150, in execute
    return_value = self.execute_callable()
  File "/home/feiyang/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 161, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/feiyang/airflow/dags/DpFreeEnergy.py", line 110, in NPT_sim
    submission.run_submission()
  File "/home/feiyang/.local/lib/python3.8/site-packages/dpdispatcher/submission.py", line 154, in run_submission
    self.upload_jobs()
  File "/home/feiyang/.local/lib/python3.8/site-packages/dpdispatcher/submission.py", line 288, in upload_jobs
    self.machine.context.upload(self)
  File "/home/feiyang/.local/lib/python3.8/site-packages/dpdispatcher/local_context.py", line 130, in upload
    raise RuntimeError('cannot find upload file ' + os.path.join(local_job, jj))
RuntimeError: cannot find upload file /home/feiyang/Documents/dptech/workdir/200K-50000bar-beta.lmp/NPT_sim/new_job/./*lmp
[2021-07-09 14:11:48,518] {taskinstance.py:1544} INFO - Marking task as FAILED. dag_id=TI_taskflow, task_id=NPT_sim, execution_date=20210709T061143, start_date=20210709T061148, end_date=20210709T061148
[2021-07-09 14:11:48,553] {local_task_job.py:151} INFO - Task exited with return code 1
```

**Comment by felix5572** [comment]
I am fixing it. And maybe try LazyLocalContext for working aroud

**Comment by Feiyang472** [comment]
> I am fixing it. And maybe try LazyLocalContext for working aroud

I tried replacing `os.path.exist` with `glob`. Seems to work.

**Comment by felix5572** [comment]
@Feiyang472  solved in pr #94
